src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,97 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.model.User;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationServiceImpl registrationService;
+    private List<User> users;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        users = new ArrayList<>();
+        
+        User validUser = new User(45534554L, "userNumberOne", "123412346", 18);
+        
+        User smallLoginUser = new User(4553712354L, "user2", "456456754", 19);
+
+        User smallPasswordUser = new User(12547646854L, "userNumberThree", "34568", 20);
+
+        User smallAgeUser = new User(2357568L, "userNumberFour", "345645745684", 16);
+
+        User nullLoginUser = new User(456356734L, null, "3465346345", 31);
+
+        User nullPasswordUser = new User(456356734L, "nullPasswordUser", null, 31);
+
+        User recurringUser = new User(34534573L, "recurringUser", "235357457", 20);
```

Comment:
**all** invalid users you should create inside test methods. You do not need them @BeforeEach
another (and best) approach - set one invalid parameter to validUser inside test methods and try to save it